### ISSUE TITLE
Forking / Switch / Routing for Flows

### ISSUE THREAD

**Issue by fawadasaurus** [candlecorp/wick#216]
**Is your feature request related to a problem? Please describe.**
We need to be able to diverge a flow based on the runtime variable values to different outcomes.

**Propose a solution**
TBD

**Describe alternatives you've considered**
???

**Additional context**
???

**Comment by fawadasaurus** [comment]
Alternate proposed solution:
```yaml

---
kind: wick/component@v1
metadata:
  version: '1'
  imports:
    - name: google_http
      component:
        kind: wick/component/composite@v1
        ref: ./recaptcha.yaml
    - name: slack_webhook
      component:
        kind: wick/component/composite@v1
        ref: ./slack.yaml
    - name: string_template_error
      with:
        template: |
          recaptcha returned {{ status }}
      component:
        kind: wick/component/composite@v1
        ref: ./template.yaml
component:
  kind: wick/component/composite@v1
  operations:
    - name: contact-form
      # Internally, a branch will be a configured instance of a core operation like `if`, e.g.
      instances:
        - name: branch
          operation: core::if
          with:
            true: self::branch1
            false: self::branch2
      flow:
        - <>.recaptcha_token -> self::recaptcha.recaptcha_token
        - self::recaptcha.status -> branch.condition
        - (<>.message,<>.email,self::recaptcha.status) -> branch.context
        - branch.output -> <>.response
      # these would be operations private to this operation.
      private:
        - name: recaptcha
          flow:
            - <>.recaptcha_token -> google_http::verify_recapcha.recaptcha_token
            - google_http::verify_recapcha.recaptcha_status -> <>.status
        - name: branch1
          flow:
            - <>.message -> slack_webhook::send_message.message
            - <>.email -> slack_webhook::send_message.email
            - slack_webhook::send_message.output -> http::response.body
            - 200 -> http::response.code
            - http::response.output -> <>.response
        - name: branch2
          flow:
            - <>.recaptcha_status -> string_template_error.status
            - string_template_error.output -> http::response.body
            - 403 -> http::response.code
            - http::response.output -> <>.response

```

**Comment by jsoverson** [comment]
Implemented in #220
